src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -136,12 +136,7 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                         fields.Add(f);
 
                         var parameterType = f.Type;
-                        if (parameterType.IsPointerOrFunctionPointer())
-                        {
-                            diagnostics.Add(ErrorCode.ERR_BadFieldTypeInRecord, f.GetFirstLocationOrNone(), parameterType);
-                            foundBadField = true;
-                        }
-                        else if (parameterType.IsRestrictedType())
+                        if (parameterType.ContainsPointerOrFunctionPointer() || parameterType.IsRestrictedType())
```

Comment:
>ContainsPointerOrFunctionPointer [](http://example.com/codeflow?start=42&length=32)

I thought we decided to allow arrays of pointers

Reply:
Thanks for pointing it out, I added a test that would have failed.